main.go

```diff
@@ -141,15 +144,39 @@ func ansibleEnable() {
 }
 
 func getAnsibleRepository(runDir string) error {
-	remoteURL := fmt.Sprintf("%s://%s", viper.GetString("http-proto"), viper.GetString("http-url"))
+	httpURL := viper.GetString("http-url")
```

Comment:
Structuring the logic as a array iteration makes it easier to expand to other providers if needed

Reply:
Right now we have two. Its not that big a change to make later, so IMHO better to keep it very straightfoward with a simple if/else and a check that an object was available.

Reply:
Okay! I added a if/else clause. Not sure what you had in mind by "an object was available". I also think it's important to make sure both variables aren't defined as this would be a breach of code contract